builder/haddock-builder.nix

```diff
@@ -0,0 +1,132 @@
+{ stdenv, buildPackages, lib, haskellLib, ghc, ghcForComponent, nonReinstallablePkgs, runCommand, writeText, writeScript, makeConfigFiles }:
+
+{ componentId
+, component
+, package
+, flags
+, revision
+, commonAttrs
+, configureFlags
+
+, doHoogle
+, hyperlinkSource
+, setupHaddockFlags
+, setupGhcOptions
+
+, needsProfiling
+, componentDrv
+, configFiles # component config files
+}:
+
+let
+  # the target dir for haddock documentation
+  docdir = docoutput: docoutput + "/share/doc/" + componentId.cname;
+
+  packageCfgDir = configFiles.packageCfgDir;
+
+  fullName = "${componentDrv.name}-haddock";
+
+  docsConfigFiles = makeConfigFiles {
```

Comment:
Worth a comment explaining how this is different from the config files for the main component.

Reply:
Added comment